scene/3d/collision_object.cpp

```diff
@@ -354,6 +374,43 @@ bool CollisionObject::get_capture_input_on_drag() const {
 	return capture_input_on_drag;
 }
 
+void CollisionObject::set_center_of_mass(const Vector3& p_center_of_mass){
+
+	center_of_mass = p_center_of_mass;
+
+	_update_shapes();
+}
+Vector3 CollisionObject::get_center_of_mass() const{
+
+	return center_of_mass;
+}
+
+Vector3 CollisionObject::get_global_center_of_mass() const {
+	return get_global_transform().translated(center_of_mass).origin;
+}
+
+void CollisionObject::calculate_center_of_mass(){
```

Comment:
I think that here we are not taking the actual mass of the collision shapes into account, what if we have two spheres one bigger than the other placed along on the same axis?

The object will be something like o O and the bouding box [o O] can be the same of [OO] but the center of mass must be different!